tensorflow/python/framework/tensor_util.py

```diff
@@ -542,10 +542,11 @@ def make_tensor_proto(values, dtype=None, shape=None, verify_shape=False,
     try:
       str_values = [compat.as_bytes(x) for x in proto_values]
     except TypeError:
-      raise TypeError(f"Failed to convert elements of {values} to Tensor. "
-                      "Consider casting elements to a supported type. See "
-                      "https://www.tensorflow.org/api_docs/python/tf/dtypes "
-                      "for supported TF dtypes.")
+      raise TypeError(f"Failed to convert elements of "
+                      "{values.__class__.__name__} to Tensor. "
+                      "Consider casting elements to a supported type. "
+                      "See https://www.tensorflow.org/api_docs/python/tf/dtypes"
+                      " for supported TF dtypes.")
```

Comment:
I am a little concerned with only including the class name of values. As the comment says, "values may be a list of objects that we could not identify a common type for".

Since the docstring says "`make_tensor_proto` accepts "values" of a python scalar, a python list, a
  numpy ndarray, or a numpy scalar.", we may validate the type of `values` in the beginning of this function and raise an early exception there. WDYT?

Reply:
> Since the docstring says "make_tensor_proto accepts "values" of a python scalar, a python list, a
numpy ndarray, or a numpy scalar.", we may validate the type of values in the beginning of this function and raise an early exception there. WDYT?

Idea looks good. But, I am not sure whether it will break any backward compatibility. If you think this will not break any backward compatibility, Please let me know the file to catch this first to raise an exception. Thanks

Reply:
@JXRiver Can you please assist on above comments from @jvishnuvardhan. Thanks!

Reply:
After another look at https://github.com/tensorflow/tensorflow/issues/50349, users wanted a clear message for lack of Keras support. I think we can leave this error message as it is and update Keras layers. I will talk to Keras developer internally. Thanks!